graph.c

```diff
@@ -225,3 +225,28 @@ edgeadddeps(struct edge *e, struct node **deps, size_t ndeps)
 	e->inorderidx += ndeps;
 	e->nin += ndeps;
 }
+
+size_t
+rootnodes(struct node **nodes)
```

Comment:
Rather than adding this `rootnodes` function, what do you think about `targets_depth` looping through nodes with `nuse == 0` directly, to avoid the intermediate array?

Reply:
I thought it would be nice to have the function available for future uses, but if it's the only use I will merge it into `targets_depth`.

Reply:
I merged the `targets` function.